CMakeLists.txt

```diff
@@ -8,51 +8,96 @@ set(CMAKE_TOOLCHAIN_FILE ${CMAKE_SOURCE_DIR}/cmake/stm32-toolchain.cmake)
 project(ATeam-Firmware
 	LANGUAGES C CXX ASM)
 
+macro(tests dir linker fname tcomment prefix)
```

Comment:
lets the make the name more descriptive "CREATE_STM32_TARGETS"
linker -> linker_file
fname -> binary_output_name
comment doesn't need to be a parameter, perhaps this can be dropped completely from the add_custom_target with the objcopy call? I don't think we need it
prefix -> stm32_arch